Automattic/jetpack
# JetpackFooter: add About, Privacy and Terms links

Fixes #31598

## Proposed changes:

The `JetpackFooter` component is used in the Jetpack flagship plugin and many other Jetpack plugins. In the former, it contains links to about, TOS, and privacy pages that are not included in the latter, though they can provide value in those instances as well. This PR integrates the _About_, _Terms_, and _Privacy_ links into the `JetpackFooter` component directly.

It also removes from the props the URL to which the Automattic logo links to since all instances using `JetpackFooter` would pass the same URL.

In more detail:
- it makes the _About_, _Terms_, and _Privacy_ links part of `JetpackFooter`
- it passes click handlers to these links as props  of `JetpackFooter`
- it removes the prop `a8cLogoHref` from `JetpackFooter`
- when applicable, it links the _Privacy_ and `An Automattic Airline` links to internal pages
- makes the Jetpack logo on the _My Jetpack_ footer match the size of the same logo in other sections

### Other information:


## Jetpack product discussion


p1687876101340969-slack-C01264051NE
p1687874518864269-slack-C02TQF5VAJD

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Make sure your site is not connected to Jetpack
- Visit the _Jetpack_ section
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `https://jetpack.com/about/`
  - Privacy: `https://automattic.com/privacy/`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `https://automattic.com/about/`
- Visit the _My Jetpack_ section
  - In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `https://jetpack.com/about/`
  - Privacy: `https://automattic.com/privacy/`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `https://automattic.com/about/`
- Connect your site to Jetpack
- In the footer, check the following links:
  - Jetpack: `https://jetpack.com/`
  - About: `https://jetpack.com/about/`
  - Privacy: `/wp-admin/admin.php?page=jetpack#/privacy`
  - Terms: `https://wordpress.com/tos/`
  - An Automattic Airline: `/wp-admin/admin.php?page=jetpack_about`
